src-native/asami/main.clj

```diff
@@ -1,51 +1,105 @@
 (ns ^{:doc "Entry point for CLI"
       :author "Paula Gearon"}
   asami.main
-  (:require [clojure.string :as s]
-            [clojure.edn :as edn]
+  (:require [asami.core :as asami]
+            [asami.graph :as graph]
             [cheshire.core :as json]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]
             [clojure.pprint :refer [pprint]]
-            [asami.core :as asami]
-            [asami.graph :as graph])
+            [clojure.string :as s])
+  (:import (java.io PushbackReader))
   (:gen-class))
 
-(def reader-opts {:readers (assoc graph/node-reader 'a/r re-pattern)})
+(def eof
+  (reify))
```

Comment:
I found this a bit confusing for a moment. I don't think it's the right approach, since the object doesn't need metadata or the `IObj` interface.

`clojure.core` generally does this using `(Object.)` (see: `get-in`, `seque`)

Reply:
This is a habit of mine from writing CLJC code to get around the annoyance of having to do the following.

```clj
#?(:clj (Object.), :cljs (js/Object.))
```